### PR TITLE
Adjust merge and categorical tests for dask-expr

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tests/test_arithmetics_reduction.py [hunk 1]
```diff
@@ -1333,7 +1341,11 @@ def test_reductions_frame_dtypes_numeric_only_supported(func):
         getattr(df, func)(numeric_only=True),
         getattr(ddf, func)(numeric_only=True),
     )
-    errors = TypeError if pa is None else (TypeError, ArrowNotImplementedError)
+    errors = (
+        (ValueError, TypeError)
+        if pa is None
+        else (ValueError, TypeError, ArrowNotImplementedError)
+    )
```
Comment (milesgranger) [inline dask/dataframe/tests/test_arithmetics_reduction.py:1348]: I'm guessing the additional error also depends on `DASK_EXPR_ENABLED`, is that right? Is it worth making this conditional as well? (My guess is no. :) )

File: dask/dataframe/tests/test_multi.py [hunk 2]
```diff
@@ -299,6 +322,8 @@ def test_sequential_joins():
 
 
 def test_merge_asof_indexed():
+    if DASK_EXPR_ENABLED:
+        pytest.skip("merge_asof not available yet in dask-expr")
```
Comment (milesgranger) [inline dask/dataframe/tests/test_multi.py:326]: [Based on pytest docs](https://docs.pytest.org/en/6.2.x/skipping.html), I think `xfail` is more appropriate in these situations.

```suggestion
@pytest.mark.xfail(DASK_EXPR_ENABLED, reason="merge_asof not available ye tin dask-expr")
def test_merge_asof_indexed():
```

**Review by milesgranger — APPROVED** [review]
One question and nit about `pytest.skip` uses is all. LGTM otherwise, thank you!

File: dask/dataframe/tests/test_arithmetics_reduction.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/tests/test_arithmetics_reduction.py:1348]: Probably not no, the tests was already ugly before this pr :)

File: dask/dataframe/tests/test_multi.py (hunk 2 above)
Comment (phofl) [reply dask/dataframe/tests/test_multi.py:326]: That doesn't work unfortunately, we can't evaluate the config in these skip contexts, don't know why

Comment (phofl) [reply dask/dataframe/tests/test_multi.py:326]: It's the decorator, inline all of them work

**Pull request merged**